### PR TITLE
Run parser tests in CI

### PR DESCRIPTION
**Pull request by malberts**
Follow-up to https://github.com/ProfessionalWiki/SPARQL/pull/13

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
@malberts seems like you unknowingly took on https://github.com/ProfessionalWiki/SPARQL/issues/6 :)

**Comment by malberts** [comment]
Right now the parser test is failing where the final HTML is not in the right order. The ordering is supposed to be done by the SPARQL query, so I'm wondering if there is an intermittent issue in the Lua in the test. However, the same test passes locally, so I don't think it's a random issue.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -69,6 +75,12 @@ jobs:
       - name: Run update.php
         run: php maintenance/update.php --quick
 
+      - name: Load Blazegraph test data
+        run: |
+          curl -X POST --data-binary @extensions/SPARQL/demoData/Q2020028.ttl --header 'Content-Type:text/turtle' http://localhost:8889/bigdata/namespace/kb/sparql
+          curl -X POST --data-binary @extensions/SPARQL/demoData/Q2291703.ttl --header 'Content-Type:text/turtle' http://localhost:8889/bigdata/namespace/kb/sparql
+          curl -X POST --data-binary @extensions/SPARQL/demoData/Q2404825.ttl --header 'Content-Type:text/turtle' http://localhost:8889/bigdata/namespace/kb/sparql
+
```
Comment (malberts) [inline .github/workflows/ci.yml:83]: We need the demo files here, so we can remove them from SPARQL-Docker.

File: src/HookHandlers.php [hunk 2]
```diff
@@ -24,6 +24,25 @@ public static function onScribuntoExternalLibraries( string $engine, array &$ext
...
+							HttpRequestFactory::CONSTRUCTOR_OPTIONS,
+							$services->getMainConfig()
+						),
+						LoggerFactory::getInstance( 'http' )
+					);
+				},
+			);
+
+			return;
+		}
```
Comment (malberts) [inline src/HookHandlers.php:44]: Annoying to have 2 big blobs like this where the only difference is the additional argument passed into HttpRequestFactory. However, something along these lines is necessary for Mw 1.39 compatibility.

File: src/SparqlLua.php [hunk 3]
```diff
@@ -5,10 +5,10 @@
 namespace ProfessionalWiki\SPARQL;
 
 use InvalidArgumentException;
-use MediaWiki\Extension\Scribunto\Engines\LuaCommon\LibraryBase;
 use MediaWiki\MediaWikiServices;
+use Scribunto_LuaLibraryBase;
 
-class SparqlLua extends LibraryBase {
+class SparqlLua extends Scribunto_LuaLibraryBase {
```
Comment (malberts) [inline src/SparqlLua.php:11]: The LibraryBase class does not exist in MW 1.39. The CS complaint is handled by https://github.com/ProfessionalWiki/SPARQL/pull/14/files#diff-0361f0c81f363476ddc6f44ab36fcbe66ee685d5f4c2a46b054924591544b766.

File: tests/parser/lua.txt [hunk 4]
```diff
@@ -91,7 +90,6 @@ return p
 !! test
 must have 3 rows and 3 columns with headers
 !! config
-wgSPARQLEndpoint = "http://blazegraph:8080/bigdata/namespace/kb/sparql"
```
Comment (malberts) [inline tests/parser/lua.txt:94]: This config should be handled by LocalSettings of the MW setup, unless it's for config that affects the outcome of the test.

**Comment by malberts** [comment]
The weird test failure with the incorrect row order was resolved after using the new test in #15.

**Pull request merged**